server/src/main/java/org/apache/uniffle/server/DefaultFlushEventHandler.java

```diff
@@ -69,6 +69,8 @@ public DefaultFlushEventHandler(
   public void handle(ShuffleDataFlushEvent event) {
     if (!flushQueue.offer(event)) {
       LOG.error("Flush queue is full, discard event: " + event);
+      // We need to release the memory when discarding the event
```

Comment:
Please add the metric like `ShuffleServerMetrics.counterTotalDroppedEventNum.inc();`

Reply:
Fixed.